### PR TITLE
chore: add missing `prettier` configs

### PR DESCRIPTION
**Pull request by kraenhansen**
## Description

A couple of packages that didn't have a .prettierrc.json config:
- ./packages/bson-transpilers
- ./packages/hadron-build
- ./packages/hadron-type-checker
- ./packages/mongodb-explain-compat

This adds the missing configs and changes to source-files reflecting that.

### Checklist
- [ ] New tests and/or benchmarks are included
- [ ] Documentation is changed or added
- [ ] I have signed the MongoDB Contributor License Agreement (https://www.mongodb.com/legal/contributor-agreement)

## Motivation and Context

- [ ] Bugfix
- [ ] New feature
- [ ] Dependency update
- [x] Misc

## Types of changes

- [ ] *Backport Needed*
- [ ] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/bson-transpilers/lib/symbol-table/shelltocsharp.js
Comment (addaleax) [inline packages/bson-transpilers/lib/symbol-table/shelltocsharp.js:1]: These are generated files, so I imagine we should add them to the prettierignore instead?

File: packages/compass/.prettierignore [hunk 1]
```diff
@@ -3,3 +3,6 @@ dist
 coverage
 build
 src/deps
+
+# Generated
+THIRD-PARTY-NOTICES.md
```
Comment (kraenhansen) [inline packages/compass/.prettierignore:8]: This got copied on my system while building the app and kept the `check` from passing locally.

File: packages/compass-e2e-tests/.prettierignore [hunk 2]
```diff
@@ -3,3 +3,7 @@
 fixtures
 .nyc_output
 coverage
+
+# Generated files
+electron-versions.json
+write-electron-versions.js
```
Comment (kraenhansen) [inline packages/compass-e2e-tests/.prettierignore:9]: These got created on my system while building the app and kept the `check` from passing locally.

**Review by Anemy — APPROVED** [review]
lgtm, changes look good, makes me happy we're aligning more things. I am a bit uncertain on some of the yaml changes (there are a lot of changes). I wonder if we want those in the prettier ignore also? Fine as is.

**Pull request merged**